core/src/test/java/org/infinispan/persistence/PersistenceCompatibilityTest.java

```diff
@@ -0,0 +1,128 @@
+import static org.testng.AssertJUnit.assertNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.util.FileLookupFactory;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.testng.annotations.Test;
+
+/**
+ * Base compatibility test for cache stores.
+ *
+ * @author Pedro Ruivo
+ * @since 11.0
+ */
+@Test(groups = "functional")
+public abstract class PersistenceCompatibilityTest<T> extends SingleCacheManagerTest {
+
+   private static final int NUMBER_KEYS = 10;
+   private final KeyValueWrapper<String, String, T> valueWrapper;
+
+   protected PersistenceCompatibilityTest(KeyValueWrapper<String, String, T> valueWrapper) {
+      this.valueWrapper = valueWrapper;
+   }
+
+   private static String key(int index) {
+      return "key-" + index;
+   }
+
+   private static String value(int index) {
+      return "value-" + index;
+   }
+
+   protected static void copyFile(String file_10_1, Path location, String fileName) throws IOException {
+      InputStream is = FileLookupFactory.newInstance()
+            .lookupFile(file_10_1, Thread.currentThread().getContextClassLoader());
+      File f = new File(location.toFile(), fileName);
+      Files.copy(is, f.toPath(), StandardCopyOption.REPLACE_EXISTING);
+   }
+
+   @Test
+   public void testReadWriteFrom101() throws Exception {
+      // 10 keys
+      // even keys stored, odd keys removed
+      beforeStartCache();
+      Cache<String, T> cache = cacheManager.getCache(cacheName());
+
+      for (int i = 0; i < NUMBER_KEYS; ++i) {
+         String key = key(i);
+         if (i % 2 != 0) {
+            assertNull("Expected null value for key " + key, cache.get(key));
+         } else {
+            assertEquals("Wrong value read for key " + key, value(i), valueWrapper.unwrap(cache.get(key)));
+         }
+      }
+
+      for (int i = 0; i < NUMBER_KEYS; ++i) {
+         if (i % 2 != 0) {
+            String key = key(i);
+            cache.put(key, valueWrapper.wrap(key, value(i)));
+         }
+      }
+
+      for (int i = 0; i < NUMBER_KEYS; ++i) {
+         String key = key(i);
+         assertEquals("Wrong value read for key " + key, value(i), valueWrapper.unwrap(cache.get(key)));
+      }
+
+   }
+
+   protected abstract void beforeStartCache() throws Exception;
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      GlobalConfigurationBuilder builder = new GlobalConfigurationBuilder().nonClusteredDefault();
```

Comment:
We can set the Global persistent location here:

`builder.globalState().persistentLocation(CommonsTestingUtil.tmpDirectory())`

That way not all of the tests have to override `amendGlobalConfigurationBuilder` and it should not negatively effect the JPA and JDBC tests that don't need it.

Reply:
:+1: